titanium-csv-import-export-tool.ts

```diff
@@ -0,0 +1,281 @@
+        let genericValidMessage = await this.validateCSV(collection, this.model);
+        let customValidMessage = await this.customValidateFn(collection);
+        this.checkAndToastErrorMessage(genericValidMessage);
+        this.checkAndToastErrorMessage(customValidMessage);
+        return {
+            isValid: genericValidMessage.isValid && customValidMessage.isValid,
+            messages: (genericValidMessage.messages || []).concat(customValidMessage.messages || []),
+            errorType: 'ParseError'
+        };
+    }
+
+    async parseFile(file: any): Promise<ICSVMessageResult> {
+        return new Promise<any>((resolve) => {
+            let options: PapaParse.ParseConfig = {
+                header: true,
+                dynamicTyping: true,
+                complete: (results: PapaParse.ParseResult, file: File) => {
+
+                    if (results.errors.length !== 0) {
+                        let errorText = '';
+                        results.errors.forEach((err) => {
+                            errorText += `${err.message}`;
+                        });
+                        resolve({ isValid: false, messages: [{ message: errorText }] } as any);
+                        return;
+                    }
+
+                    this.importCollection = results.data;
+                    resolve({ isValid: true } as any);
+                },
+                error: (err: any) => {
+                    resolve({
+                        isValid: false,
+                        messages: [{ message: err.message }],
+                        errorType: 'ApplicationError'
+                    } as any);
+                }
+            };
+
+            Papa.parse(file, options);
+        });
+    }
+
+    @observe('filename')
+    filenameChanged() {
+        if (!this.filename)
+            this.filename = 'export';
+    }
+
+    checkAndToastErrorMessage(csvMessage: ICSVMessageResult): ICSVMessageResult {
+        if (csvMessage.errorType === 'ApplicationError') {
+            let messages = (csvMessage.messages || []).map(function (entry) {
+                return entry.message;
+            });
+            let toastMessage = messages.join('/n');
+            this.toast.show({ text: toastMessage });
+            csvMessage.messages = [];
+            return csvMessage;
+        }
+        return csvMessage;
+    }
+
+    rawCSVtoCollection(rawCSV: string[][]): any[] {
+        let keys = rawCSV[0];
+        let collection = [];
+
+        return rawCSV.slice(1).map((eachRawCSVRow: string[], rowNumber: number) => {
+            let returnRow: { [index: string]: string | number | undefined } = {};
+            keys.forEach((eachKey: string, keyIndex: number) => {
+                if (!eachRawCSVRow[keyIndex]) {
+                    returnRow[eachKey] = undefined;
+                    return;
+                }
+                if (parseInt(eachRawCSVRow[keyIndex]).toString() === eachRawCSVRow[keyIndex]) {
```

Comment:
What's the purpose of parsing the cell to an int, making it a string and comparing it with its original value? Are you verifying it's a valid int? If so, 
`isNan(parseInt(field, 10))`
is probably what you're after.

Also keep in mind parseInt's big brother, parseFloat which is always base 10 and handles decimals.